Fix GTK+3 build further and make it build with CMake too

adriweb opened the pull request:
- Remove some deprecated stuff that don't work anymore
- Force some features to look like GTK2 (button labels, menu item icons)

So far, seems to be working fine on macOS at least:
![Screenshot](https://i.imgur.com/aoPugFq.jpg)

adriweb commented:
Added the CMake build option to use GTK+3 on TiLP, and the appropriate corresponding build matrix items on Travis

debrouxl commented on .travis.yml:
```diff
@@ -1,42 +1,72 @@
-        - git
-        - autoconf
-        - automake
-        - autopoint
```
These should remain, just in case: here on Debian sid amd64, neither build-essential, nor libtool, nor gettext have direct, or obvious indirect, dependencies on these three packages.

adriweb replied:
Sure!

adriweb commented:
Rebased, but modifications still needed:

1. build option like `BUILD_USING_GTK_VERSION=[234]`

2. compat stuff:
```
17:27:58 <@debroux_> Adriweb: dans ton hunk du premier commit qui touche tilp/trunk/src/tilp.c , il faut ajouter...
17:28:16 <@debroux_> #if ! GTK_CHECK_VERSION(3,10,0) / #endif
17:28:35 <@debroux_> de part et d'autre du bloc d'appels à gtk_image_menu_item_set_always_show_image().
```

adriweb commented:
Latest pushes:
- `try_static_libs.cmake` was improved to have less hardcoded things and work correctly in more cases (the recent macOS Big Sur changes made me implement cleaner/proper lib detection!). This will have to be pushed on the tilibs side too.
- Fixed some other potential cmake issues
- Added `GTK_CHECK_VERSION` for `gtk_image_menu_item_set_always_show_image` since that's deprecated.
